metagraph/integration_tests/test_taxonomy.py

```diff
@@ -0,0 +1,165 @@
+import unittest
+import subprocess
+from subprocess import PIPE
+from parameterized import parameterized
+
+from tempfile import TemporaryDirectory
+import os
+
+
+"""Test taxonomy classification workflow"""
+
+METAGRAPH = './metagraph'
+PROTEIN_MODE = os.readlink(METAGRAPH).endswith("_Protein")
+TEST_DATA_DIR = os.path.dirname(os.path.realpath(__file__)) + '/../tests/data'
+TAX_DATA_DIR = TEST_DATA_DIR + "/taxonomic_data"
+
+tax_tests = {
+    'one_thread': {
+        'threads': 1,
+        'discovery_fraction': 0,
+    },
+    '5_threads': {
+        'threads': 5,
+        'discovery_fraction': 0,
+    },
+    '16_threads': {
+        'threads': 16,
+        'discovery_fraction': 0,
+    }
+}
+
+test_params = [name for name, _ in tax_tests.items()]
+
+class TestTaxonomy(unittest.TestCase):
+    def setUp(self):
+        self.tempdir = TemporaryDirectory()
+        self.tax_parent = {}
+        self.tax_root = -1
+        tax_lines = open(TAX_DATA_DIR + '/dumb_nodes.dmp').readlines()
+        for line in tax_lines:
+            act_node = line.split('\t')[0].strip()
+            act_parent = line.split('\t')[2].strip()
+            self.tax_parent[act_node] = act_parent
+            if act_node == act_parent:
+                self.tax_root = act_node
+
+    def is_descendant(self, node: str, query: str) -> bool:
+        while query != self.tax_root:
+            query = self.tax_parent[query]
+            if query == node:
+                return True
+        return False
+
+    @parameterized.expand(test_params)
+    @unittest.skipIf(PROTEIN_MODE, "No canonical mode for Protein alphabets")
```

Comment:
I am not very sure if we need to skip this test or not. This integration test is using only some viral sequences stored at `tests/data/taxonomic_data/` and no any proteins